File: javascript/对象/遍历对象/keys.js
// keys();

// 语法

// Object.keys(obj);

// 参数说明

// object 必须 => 操作的对象

// 详情

// 返回一个数组，包含对象的可枚举的实例属性名称

let obj = {
    name: "Tom",
    age: 18,
    getName: function(){
        console.log(`${this.name}`);
    }
}

// 1) getOwnPropertyNames 与 keys 方法返回的内容都相同

console.log(Object.getOwnPropertyNames(obj)); // => ["name", "age", "getName"] ：返回对象的所有实例成员
console.log(Object.keys(obj)); // => ["name", "age", "getName"] ：返回对象的所有可枚举成员

// 设置对象的name属性不可枚举

Object.defineProperty(obj, 'name', {
    enumerable: false
});

// 2)keys方法，只包含可枚举成员

console.log(Object.getOwnPropertyNames(obj)); // => ["name", "age", "getName"] ：返回对象的所有实例成员
console.log(Object.keys(obj)); // => ["age", "getName"] ：返回对象的所有可枚举成员

File: python/字典(对象)/遍历字典/keys.py
# 字典 keys() 方法返回一个可迭代对象，可以使用 list() 来转换为列表。

# dict.keys()

# 返回值 返回一个迭代器。

Object = {
    "name": "Tom",
    "age": 18,
    "other": "其他"
}

print(Object.keys()) # dict_keys(['name', 'age', 'other'])

for i in Object.keys():
    print('key', i) # key name\key age\key other

print(list(Object.keys())) # ['name', 'age', 'other']